scripts/nb-tester/qiskit_docs_notebook_tester/__init__.py

```diff
@@ -37,20 +38,70 @@
 matplotlib.set_loglevel("critical")
 """
 
-# If not submitting jobs, we also run this code before notebook execution to mock the real backend
-MOCKING_CODE = """\
-import warnings
-from qiskit_ibm_runtime import QiskitRuntimeService
-from qiskit.providers.fake_provider import GenericBackendV2
+def render_kwarg(arg: str, val: any):
+    if isinstance(val, str):
+        return f"{arg}=\"{val}\""
+    return f"{arg}={val}"
+
+def render_kwargs(kwargs: dict[str, any]):
+    return ", ".join(render_kwarg(arg, val) for arg, val in kwargs.items())
 
-def patched_least_busy(self, *args, **kwarg):
-  return GenericBackendV2(num_qubits=6, control_flow=True)
+def generate_backend_patch(
+    backend_name: str, 
+    provider: Literal["qiskit-ibm-runtime", "qiskit-fake-provider", "runtime-fake-provider"] = "qiskit-ibm-runtime", 
+    generic_backend_kwargs: dict[str, any] = None, 
+    runtime_service_kwargs: dict[str, any] = None
+) -> str:
+    """
+    Generate code for fetching a custom backend to inject into a notebook.
+    """
 
-QiskitRuntimeService.least_busy = patched_least_busy
+    patch = dedent("""
+    from qiskit_ibm_runtime import QiskitRuntimeService
+    import warnings
+
+    warnings.filterwarnings("ignore", message="Options {.*} have no effect in local testing mode.")
+    warnings.filterwarnings("ignore", message="Session is not supported in local testing mode or when using a simulator.")
+    """)
+
+    if provider == "qiskit-fake-provider":
+        if not generic_backend_kwargs:
+            generic_backend_kwargs = {}
+        qiskit_fake_provider_args = render_kwargs(generic_backend_kwargs)
+        patch += dedent(f"""
+        from qiskit.providers.fake_provider import GenericBackendV2
+        def patched_least_busy(self, *args, **kwargs):
+            return GenericBackendV2({qiskit_fake_provider_args})
+        """)
+
+    elif provider == "runtime-fake-provider":
+        patch += dedent(f"""
+        from qiskit_ibm_runtime.fake_provider import FakeProviderForBackendV2
+
+        def patched_least_busy(self, *args, **kwargs):
+            provider = FakeProviderForBackendV2()
+            return provider.backend("{backend_name}")
+        """)
+
+    elif provider == "qiskit-ibm-runtime": 
+        # Generates a set of arguments for QiskitRuntimeService using kwargs
+        if not runtime_service_kwargs:
+            runtime_service_kwargs = {}
+        qiskit_runtime_service_args = render_kwargs(runtime_service_kwargs)
```

Comment:
We prefer self-documenting code over comments, unless when the comment adds something you can't easily infer

```suggestion
        qiskit_runtime_service_args = render_kwargs(runtime_service_kwargs or {})
```